src/ircode.jl

```diff
@@ -48,24 +48,39 @@ Return the `IRCode` object along with inferred return type.
 function code_ircode_by_mi(f, mi::MethodInstance; world=get_world_counter(), interp=NativeInterpreter(world))
     return typeinf_lock() do
         result = Core.Compiler.InferenceResult(mi)
-        frame = Core.Compiler.InferenceState(result, false, interp)
+        @static if VERSION < v"1.8-"
+            frame = Core.Compiler.InferenceState(result, false, interp)
+        else
+            frame = Core.Compiler.InferenceState(result, :local, interp)
+        end
         frame === nothing && return nothing
 
         if typeinf(interp, frame)
             opt_params = OptimizationParams(interp)
-            opt = OptimizationState(frame, opt_params, interp)
-            preserve_coverage = coverage_enabled(opt.mod)
-            ci = opt.src; nargs = opt.nargs - 1;
-            ir = convert_to_ircode(ci, copy_exprargs(ci.code), preserve_coverage, nargs, opt)
-            ir = slot2reg(ir, ci, nargs, opt)
+            sv = OptimizationState(frame, opt_params, interp)
+            preserve_coverage = coverage_enabled(sv.mod)
+            @static if VERSION < v"1.8-"
+                ci = sv.src; nargs = sv.nargs - 1;
+                ir = convert_to_ircode(ci, copy_exprargs(ci.code), preserve_coverage, nargs, sv)
+                ir = slot2reg(ir, ci, nargs, sv)
+            else
+                ci = sv.src
+                ir = convert_to_ircode(ci, sv)
+                ir = slot2reg(ir, ci, sv)
+            end
             # passes
-            ir = f(ir, opt)
-            opt.src.inferred = true
+            ir = f(ir, sv)
```

Comment:
@aviatesk `f` is used here, it kinda allow custom `IRCode` transform from this reflection function since the `run_passes` function is not called here and is replaced by the `default_julia_passes` function, so that I can put my own passes to test with